### ISSUE TITLE
Prettier 3.x Plugin Loading New Behavior

### ISSUE THREAD

**Issue by samypr100** [diffplug/spotless#1798]
If you are submitting a **bug**, please include the following:

- [x] summary of problem
- [x] gradle or maven version: Gradle
- [x] spotless version: 6.21.0
- [x] operating system and version: Ubuntu 22.04

---

Hello, Prettier v3.x [changed](https://prettier.io/blog/2023/07/05/3.0.0.html#plugin-search-feature-has-been-removed-14759httpsgithubcomprettierprettierpull14759-by-fiskerhttpsgithubcomfisker) the way it load plugins. Now plugins are not loaded by default anymore, instead users have to manually pass a `--plugin=prettier-plugin-name`, or configure it in a configuration file via  `plugins: ['prettier-plugin-name']`.

I tried something like this to see if it would work:

```groovy
spotless {
    // ...
    java {
        // ...
        prettier(['prettier': '3.0.2', 'prettier-plugin-java': '2.3.0'])
                .config(['plugins': ['prettier-plugin-java'], 'parser': 'java', 'tabWidth': 4, 'printWidth': 120, 'useTabs': false])
        // ...
    }
}
```

But gave me the error:

```
Only values of type 'String', 'JsonRawValue', 'Number' and 'Boolean' are supported. You provided: [java, [prettier-plugin-java], 120, 4, false]
```

Am I missing something? I saw prettier v3 support was added in https://github.com/diffplug/spotless/pull/1760, but I didn't see any noticeable docs updates about the plugin loading changes.

Note, just by doing

```groovy
prettier(['prettier': '3.0.2', 'prettier-plugin-java': '2.3.0'])
  .config(['parser': 'java', 'tabWidth': 4, 'printWidth': 120, 'useTabs': false])
```

The following error happens:

```
Unexpected response status code at /prettier/format [HTTP 500] -- (Error while formatting: ConfigError: Couldn't resolve parser "java".)
```

Which is why I tried to go the manual plugin loading route.

**Comment by simschla** [comment]
Thanks for reporting. I must have missed this in the docs. Will look into it!

**Comment by nedtwigg** [comment]
Shipped in `plugin-gradle 6.22.0` and `plugin-maven 2.40.0`.
